lib/bitcoin/lock/LockResolver.ts

```diff
@@ -151,4 +153,14 @@ export default class LockResolver {
       throw BitcoinError.createFromError(ErrorCode.LockResolverTransactionNotFound, e);
     }
   }
+
+  private async calculateLockStartingBlock (transaction: BitcoinTransactionModel): Promise<number> {
+    if (transaction.confirmations === 0 || transaction.confirmations < 0) {

```

Comment:
Sanity: Maybe I am misreading this, can confirmations be less than zero?
```suggestion
    if (transaction.confirmations <= 0) {

```

Reply:
They cannot be but just for sanity doing the `<` check.

During integration, I saw that the `confirmations` was undefined for the broadcasted (and not yet written) transactions. So I had the `===` check. I can change it to the `<=` because I did update the `bitcoinclient` to make sure that `confirmations` is never undefined.